### PR TITLE
Remove another DefiningAnchor::Bubble usage

### PR DESCRIPTION
**Pull request by oli-obk**
This doesn't affect anything, but removes one difference between the new and the old solver.

r? @lcnr 

cc https://github.com/rust-lang/rust/issues/113278

### CODE REVIEW

**Comment by compiler-errors** [comment]
Those failing tests were the ones I was talking about -- we could bump the logic in `compiler/rustc_lint/src/opaque_hidden_inferred_bound.rs` to be a hard error for TAITs in order to keep those tests failing.

**Comment by oli-obk** [comment]
oof... yea, at least all the cases are now hard errors, instead of just some 😆

File: compiler/rustc_lint/src/opaque_hidden_inferred_bound.rs [hunk 1]
```diff
@@ -140,25 +140,51 @@ impl<'tcx> LateLintPass<'tcx> for OpaqueHiddenInferredBound {
...
+                                OpaqueHiddenInferredBoundLint {
+                                    ty,
+                                    proj_ty: proj_term,
+                                    assoc_pred_span,
+                                    add_bound,
+                                },
+                            );
+                        }
+                        hir::OpaqueTyOrigin::TyAlias { .. } => {
+                            cx.tcx.sess.emit_err(OpaqueHiddenInferredBoundErr {
```
Comment (lcnr) [inline compiler/rustc_lint/src/opaque_hidden_inferred_bound.rs:160]: hard errors in `rustc_lint` is somewhat :/

I also don't think we *should* error (or even lint, for that matter) here, tho

Comment (lcnr) [reply compiler/rustc_lint/src/opaque_hidden_inferred_bound.rs:160]: opened a zulip thread about this https://rust-lang.zulipchat.com/#narrow/stream/315482-t-compiler.2Fetc.2Fopaque-types/topic/nested.20opaque.20bounds

Comment (compiler-errors) [reply compiler/rustc_lint/src/opaque_hidden_inferred_bound.rs:160]: From a technical perspective, yeah, we shouldn't be doing hard errors inside of the lint infrastructure I think.

**Comment by oli-obk** [comment]
This has become obsolete.

**Pull request closed without merging**